### ISSUE TITLE
Add marketplace Vscode plugin for Puppet seems not working

### ISSUE THREAD

**Issue by disaster37** [eclipse-che/che#14099]
Hi,
I currently use the lastest version of eclipse CHE 7.0.0-rc-4.0-SNAPSHOT and I try to use the Vscode plugin for puppet https://marketplace.visualstudio.com/items?itemName=jpogran.puppet-vscode.

I have following this article https://github.com/eclipse/che/wiki/CHE-7---Development-Workflow to do that, but it seems doesn't work.

The workspace run fine, the docker container that handle the puppet-vscode plugin run fine and download the vsix file, but it seems that theia never load it. In theia, on plugin view, the plugin seems not installed (I look the plugin with button install) and the code highlighting doesn't work.

Sample of devfile to reproduce it:
```
metadata:
  name: puppet
components:
  - type: chePlugin
    alias: puppet-plugin
    reference: >-
      https://raw.githubusercontent.com/disaster37/che-remote-plugin-puppet/f2930ce56c0c7322261425c6152ffdbc597dcc52/plugins/jpogran/puppet-vscode/latest/meta.yaml
  - mountSources: true
    memoryLimit: 512Mi
    type: dockerimage
    volumes:
      - name: userspace
        containerPath: /home/dev
    image: 'webcenter/che-puppet:latest'
    alias: puppet-cli
apiVersion: 1.0.0
```

Anybody know what is the mistake ?

**Comment by ibuziuk** [comment]
@disaster37 could you please add screenshots of the plugin view and logs from `webcenter/che-puppet:latest` container ?

**Comment by disaster37** [comment]
For webcenter/che-puppet:latest ,on docker output, there are nothing.
For webcenter/che-remote-plugin-puppet:latest, on docker output, there are `Theia Endpoint 1/pid listening on port 2503`

And the folder /workspace_logs is empty. 

To finish, the screencapture of theiga:
[https://pasteboard.co/IqHG9PI.png](url)

**Comment by l0rd** [comment]
@evidolob can you have a look please?

**Comment by disaster37** [comment]
Hi,

I finnaly found a part of mistake. I replace my base image from `wsskeleton/theia-endpoint-runtime` to `eclipse/che-theia-endpoint-runtime` and now the plugin is enabled. So the documentation is not right, maybee you can fix it.

Now I found another problème because of the command provided by vscode plugin not exist.

**Comment by l0rd** [comment]
@disaster37 thanks for reporting the documentation problem I have created the PR https://github.com/eclipse/che-docs/pull/772

**Comment by l0rd** [comment]
Can you be more specific about the new error message that you get?

**Comment by disaster37** [comment]
I doesn't look any error, but it seems that only highlighting code that work.
All other feature seems not work, like:
  - Auto completion
  - outline info
  - Code generator
  - etc.

How I can found error or troubleshoot that ?

**Comment by disaster37** [comment]
Hi,

I have found another mistake with image `eclipse/che-theia-endpoint-runtime`. There are no user with uid 1724, so when the container plugin is lauch, the user not really exist and it break puppet cli that use by the plugin.

I need to add this on my dockerfile, and maybee it can be added directly on your image to avoid this kind of behavior:

```
ENV \
    USER=theia \
    GROUP=theia \
    USER_ID=1724 \
    GROUP_ID=1724

RUN \
    addgroup -g ${GROUP_ID} ${GROUP} &&\
    adduser -s /bin/sh -G ${GROUP} -D -u ${USER_ID} ${USER}

WORKDIR /projects
```


But after that, puppet cli work fine (I open terminal on container from theiga and I can run puppet cli), but the plugin seems never lauch puppet cli (autocompletion, doc, block generator not work).

I have tried to install vscode on my workstation, install puppet from gem like on my container and configure the plugin like in theiga, and all work fine ...

```json
{
   "puppet.installType": "agent",
   "puppet.installDirectory": "/usr/bin"
}
```
